pkg/controller/servicebinding/secret.go

```diff
@@ -27,10 +32,68 @@ func (s *secret) buildResourceClient() dynamic.ResourceInterface {
 	return s.client.Resource(gvr).Namespace(s.ns)
 }
 
-// createOrUpdate will take informed payload and either create a new secret or update an existing
-// one. It can return error when Kubernetes client does.
-func (s *secret) createOrUpdate(payload map[string][]byte, ownerReference metav1.OwnerReference) (*unstructured.Unstructured, error) {
+// compare existing secret and new secret
+func (s *secret) isSame(secretName string, payload map[string][]byte) bool {
 	logger := s.logger.WithValues("Namespace", s.ns, "Name", s.name)
+
+	resourceClient := s.buildResourceClient()
+	existingSecret, err := resourceClient.Get(secretName, metav1.GetOptions{})
+	if err != nil {
+		logger.Error(err, "Error fetching secret")
+		return false
+	}
+	existingSecretData, _, _ := unstructured.NestedMap(existingSecret.Object, "data")
+
+	payloadInterim := make(map[string]interface{})
+	for k, v := range payload {
+		payloadInterim[k] = base64.StdEncoding.EncodeToString(v)
+	}
+
+	comparisonResult := nestedMapComparison(existingSecretData, payloadInterim)
+
+	if comparisonResult.Success {
+		logger.Debug("Secret data is same.")
+		return true
+	}
+
+	return false
+}
+
+func sortKeys(payload map[string][]byte) string {
+	keys := make([]string, 0, len(payload))
+	for k := range payload {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	var text []string
+	for _, k := range keys {
+		s := fmt.Sprintf("%s%s", k, string(payload[k]))
+		text = append(text, s)
+	}
+	return strings.Join(text, "")
+}
+
+func hash(text string) (string, error) {
```

Comment:
Are you planning to include unit tests for `hash`?

Reply:
That function has been removed.
I have added test for uniqueSecretName() method -> c3fe49f